### PR TITLE
refactor spend function and add error to result

### PR DESCRIPTION
**Pull request by therealssj**
Refactor gui/wallet.go:Spend function 

return `balance`, `transaction`, `error` instead of `SpendResult`
also added some documentation

Resolves: #749

### CODE REVIEW

File: src/gui/wallet.go [hunk 1]
```diff
@@ -23,46 +23,44 @@ type SpendResult struct {
 	Error       string                     `json:"error,omitempty"`
 }
 
-// Spend spend coins from specific wallet
+// Spend spends coins from specific wallet
 func Spend(gateway *daemon.Gateway, walletID string, coins uint64, dest cipher.Address) *SpendResult {
```
Comment (gz-c) [inline src/gui/wallet.go]: Return `(balance, transaction, error)` from this method
The caller of this should pack it into the `SpendResult` json struct

File: src/gui/wallet.go [hunk 2]
```diff
@@ -23,50 +23,55 @@ type SpendResult struct {
...
-		}
 
-		var txStr string
-		txStr, err = visor.TransactionToJSON(*tx)
-		if err != nil {
-			break
-		}
-
-		logger.Info("Spend: \ntx= \n %s \n", txStr)
+	if tx, err = gateway.Spend(walletID, coins, dest); err != nil {
```
Comment (gz-c) [inline src/gui/wallet.go]: write it as 

```go
tx, err := gateway.Spend(...)
if err != nil { 
    ...
    return nil, nil, err 
}
```

Remove the `var` declarations at the top

Comment (therealssj) [reply src/gui/wallet.go]: `return nil, nil, err` ?
Should I not return the wallet balance and transaction when it fails?

Comment (gz-c) [reply src/gui/wallet.go]: No, you should return empty values for the initial arguments, when there is an error

Comment (gz-c) [reply src/gui/wallet.go]: No, you should return empty values for the initial arguments, when there is an error

**Pull request merged**